assets/weapons/tags/schemas/base_weapon.json

```diff
@@ -0,0 +1,122 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Base Weapon",
+    "description": "These are properties shared by all weapon types.",
+    "type": "object",
+    "properties": {
+        "$schema": {
+            "type": "string"
+        },
+        "type": {
+            "enum": [
+                0,
+                1,
+                2,
+                3
+            ],
+            "description": "0 = melee, 1 = raycast, 2 = projectile, 3 = special"
+        },
+        "name": {
+            "type": "string",
+            "description": "The name of the weapon. Used both for identification and display."
+        },
+        "fireRate": {
+            "type": "number",
+            "description": "The rate of fire in units of rounds per second"
+        },
+        "bullets": {
+            "type": "integer",
+            "description": "The number of rounds per magazine"
+        },
+        "ammo": {
+            "type": "integer",
+            "description": "TODO"
```

Comment:
Anything with a TODO description is something I wasn't sure of. I didn't dig around to infer usage so we could discuss first.